pkg/skaffold/survey/survey.go

```diff
@@ -74,20 +90,32 @@ func isSurveyPromptDisabled(configfile string) (*sConfig.GlobalConfig, bool) {
 		*cfg.Global.Survey.DisablePrompt
 }
 
-func recentlyPromptedOrTaken(cfg *sConfig.GlobalConfig) bool {
+func (s *Runner) recentlyPromptedOrTaken(cfg *sConfig.GlobalConfig) bool {
 	if cfg == nil || cfg.Global == nil || cfg.Global.Survey == nil {
-		return false
+		return s.taken(cfg.Global.Survey)
```

Comment:
This will result in a nil panic given the `if` condition above.

Reply:
s.taken first line has a check to make sure `cfg.Global.Survey` is nil and a test is added.

Reply:
update: not `s.taken` but function calls below. There is a nil input test
```
func (s *Runner) taken(gc *sConfig.SurveyConfig) bool {
	// fetch candidate surveys not taken by the user.
	s.surveyID = s.selectSurvey(gc)
	return s.surveyID == ""
}

func (s *Runner) selectSurvey(gc *sConfig.SurveyConfig) string {
	taken := surveysTaken(gc)
...
}

func surveysTaken(sc *sConfig.SurveyConfig) map[string]struct{} {
	if sc == nil {
		return map[string]struct{}{}
}

```

Reply:
hmm. Integration tests failed due to npe. 
I am still scratching my brain why this happened. 

Made some changes.